### DISCUSSION TITLE
Type (Model) Object: Field whose value is generated by a model class method

### DISCUSSION THREAD

**Discussion by nimmolo** [rmosolgo/graphql-ruby#3688] [Q&A] [answered]
My Rails DB has a model `Image`. (It predates, and does not use, Active Storage, but it's similar in some ways - images can belong to other models: `User`, `Observation`, `Publication`, `GlossaryTerm`, etc.) 

But the image URLs are not in the database -- they're generated dynamically in Rails by a method of the `Image` class, `Image.url(size, id)`. Over the years image storage has moved around, and the method allows for moving them with minimal disruption.

I'd like to be able to return those image URLs via GraphQL. Is there a way to serve a field whose value is generated dynamically? I've found a few issues and SO threads that seem to refer to this, but it's not clear to me how a field can call a method outside GraphQL. If I call `Image.url(:thumbnail, object.id)` in the field resolver method, it's looking for that method on my `Types::Image`, not the `Image` model.

So far, my GraphQL fields come directly from the database: :id, :notes, :created_at, etc.

**Comment by nimmolo** [comment, marked as answer]
Oof! 🤭🙃 so that's why there is that convention to name your type ImageType... to avoid class name conflicts! 
I figured the namespace would take care of it but not here.

Problem solved.

**Comment by rmosolgo** [comment]
In Ruby, you can explicitly reference top-level constants by prefixing them with `::`. For example, `::Image` would always reference the top-level Image class (the model), and never `Types::Image`. I think if you used `::Image.url(...)`, that would also solve the problem. (But you're right, that's why the `*Type` suffix has stuck around.)
